packages/clients/textLocator/src/plugins/GeometrySearch/store/actions/watchers.ts

```diff
@@ -0,0 +1,104 @@
+import { PolarStore } from '@polar/lib-custom-types'
+import { GeometrySearchGetters, GeometrySearchState } from '../../types'
+import {
+  TitleLocationFrequency,
+  searchLiteratureByToponym,
+} from '../../../../utils/literatureByToponym'
+
+export function setupWatchers(
+  this: PolarStore<GeometrySearchState, GeometrySearchGetters>,
+  { dispatch, rootGetters }
+) {
+  // load titleLocationFrequency on each featureCollection update
+  this.watch(
+    () => rootGetters['plugin/geometrySearch/featureCollection'],
+    () => dispatch('updateFrequencies')
+  )
+}
+
+const requestLiteraturePerFeature = (
+  featureCollection: GeometrySearchState['featureCollection'],
+  textLocatorBackendUrl: string
+) =>
+  featureCollection.features
+    .map((feature) => feature.properties?.names?.map((name) => name.Name))
```

Comment:
```suggestion
    .map((feature) => feature.properties.names.map((name) => name.Name))
```
Based on the type they are supposed to always be present. Is this correct?